src/OpenRiaServices.Tools/Framework/DataContractProxyGenerator.cs

```diff
@@ -536,8 +536,12 @@ private IEnumerable<Attribute> FilterTypeAttributes(AttributeCollection typeAttr
             }
 
             // Filter out attributes in filteredAttributes as well as DataContractAttribute and KnownTypeAttribute (since they are already handled)
-            return typeAttributes.Cast<Attribute>().Where(a => a.GetType() != typeof(DataContractAttribute) && a.GetType() != typeof(KnownTypeAttribute) && 
-                !(filteredAttributes.Contains(a)));
+            return typeAttributes.Cast<Attribute>().Where(a => a is not (DataContractAttribute or KnownTypeAttribute
+#if NET8_0
+                // Filter out NullableContextAttribute, should only be used by compiler
+                or System.Runtime.CompilerServices.NullableContextAttribute
```

Comment:
If we decide to only compile tools for .NET 6 then this check should be based on type name `a.GetType() == "System.Runtime.CompilerServices.NullableContextAttribute"` (and it should be changed to `#if NET`

nit: kod is perfectly fine , but it might check whole class hierarky instead of just comparing type.

Reply:
Just a heads up: 

1. I got this problem whit current codegen and having .net8 installed.  I manually changed .runtimeconfig for codegen.exe in my nugey folder to use .net 7 to solve it temporarily

2. There are similar issues with attributes for nullable reference types "string?" And required properties (but we should probably create separate issue and PR to solve that)

Reply:
@SandstromErik I added bullets points to #315 about the attributes